app/src/main/java/com/andrew/tasky/agenda/domain/AgendaNotificationScheduler.kt

```diff
@@ -0,0 +1,8 @@
+package com.andrew.tasky.agenda.domain
+
+import com.andrew.tasky.agenda.domain.models.AgendaNotificationInfo
+
+interface AgendaNotificationScheduler {
+    fun schedule(agendaNotificationInfo: AgendaNotificationInfo)
+    fun cancel(agendaNotificationInfo: AgendaNotificationInfo)
```

Comment:
Wouldn't it be enough to just pass the ID of the agenda item here? That's all you need to get its hashcode and cancel the notification.